### DISCUSSION TITLE
about domains, tab-bar

### DISCUSSION THREAD

**Discussion by fedoranvar** [wezterm/wezterm#1724] [Q&A]
Good day to you!

1. How to display current domain name in status bar?
2. How to quit or detach domain by keybinding?
3. How to toggle tab-bar by keybiidning?
4. How i can apply settings according to current domain?
(for examplle,   custom keybindindings for specified domains)

Thank you!

**Comment by wez** [comment]
> Good day to you!

And to you!

> 1. How to display current domain name in status bar?
> 2. How to quit or detach domain by keybinding?

We don't currently expose domain information, or have a well-developed detach concept.
https://github.com/wez/wezterm/issues/1531 added a workspaces concept that I think you might want to look at instead.

[window:active_workspace()](https://wezfurlong.org/wezterm/config/lua/window/active_workspace.html) to show the active workspace and [SwitchToWorkspace](https://wezfurlong.org/wezterm/config/lua/keyassignment/SwitchToWorkspace.html) for switching workspaces.

> 3. How to toggle tab-bar by keybiidning?

Take a look at the examples on [window:set_config_overrides()](https://wezfurlong.org/wezterm/config/lua/window/set_config_overrides.html) that show how to change other config settings in response to pressing keys.  You could adapt those to operate on `enable_tab_bar` instead.

> 4. How i can apply settings according to current domain?
>    (for examplle,   custom keybindindings for specified domains)

Because key assignments are scoped to the window and domains are per-pane, there's a bit of a tricky mismatch with this today.

If you're using workspaces, you could potentially use the [window-config-reloaded](https://wezfurlong.org/wezterm/config/lua/window-events/window-config-reloaded.html) event to determine the current workspace and then use  [window:set_config_overrides()](https://wezfurlong.org/wezterm/config/lua/window/set_config_overrides.html) to adjust the key assignments.

You might have to manually trigger a config reload (eg: CTRL-SHIFT-R) to notice that change though.

**Comment by fedoranvar** [comment]
@wez 

one more question:

I've defined two custom domains with different names and on different sockets-path, but if i launched one domain1 and after that i laucnehd `wezterm connect domain2' it shows me domain1
 
can multiple domains work together on host machine?

**Reply by wez** [reply]
If you're running two separate mux servers, you should be able to connect to them with separate `wezterm connect` invocations.
However, the workspace feature is intended to make things simpler: just run a single mux server and then use workspaces instead of multiple domains
